packages/base/src/dialogs/symbology/vector_layer/types/Graduated.tsx

```diff
@@ -171,7 +171,9 @@ const Graduated = ({
 
     layer.parameters.symbologyState = symbologyState;
     layer.parameters.color = newStyle;
-    layer.type = 'VectorLayer';
+    if (layer.type === 'HeatmapLayer') {
+      layer.type = 'VectorLayer';
```

Comment:
That explains the `layer.type` assignment we used to have 👍🏽

Reply:
Yes, one concern remains though that this is not ideal if we apply heatmapLayer to any other type than `VectorLayer` for example `VectorTileLayer` will fail like before if we enable heatmaplayer symbology on it. We'd need to find a better solution.

However it's not concerning as of now since we apply heatmaplayer to `VectorLayer` currently.